clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java

```diff
@@ -661,10 +661,20 @@ public void testNetworkThreadTimeRecorded() throws Exception {
         assertEquals(0, selector.completedReceives().size());
 
         selector.unmute(node);
-        while (selector.completedReceives().isEmpty()) {
-            selector.poll(100L);
-            assertEquals(0, selector.numStagedReceives(channel));
-        }
+        // Wait for echo server to send the message back
+        TestUtils.waitForCondition(new TestCondition() {
+            @Override
+            public boolean conditionMet() {
+                try {
+                    selector.poll(100L);
+                    assertEquals(0, selector.numStagedReceives(channel));
+                } catch (IOException e) {
+                    return false;
+                }
+                return selector.completedReceives().size() > 0;
```

Comment:
Could use !`isEmpty()` instead of `.size()`?

Reply:
@rajinisivaram  Thanks for the review.  test is failing most of the times.   will create JIRA with relevant logs.